### PR TITLE
[WIP] (OLD : see #1122 and #1123) DC Area Interchange Control Outer Loop

### PR DESCRIPTION
**Pull request by vmouradian**
HAS BEEN SPLITTED INTO 2 PRs :  

#1122 
#1123 

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

feature

**What is the current behavior?**

AIC not supported for DC Loadflow

**What is the new behavior (if this is a feature change)?**
AIC outerloop is supported for DC Loadflow

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/test/java/com/powsybl/openloadflow/ac/AreaInterchangeControlTest.java [hunk 1]
```diff
@@ -107,19 +150,38 @@ void remainingMismatchDistributeOnReferenceGenerator() {
...
+
+    private void remainingMismatchThrow() {
         parametersExt.setSlackDistributionFailureBehavior(OpenLoadFlowParameters.SlackDistributionFailureBehavior.THROW);
         Network network = MultiAreaNetworkFactory.createOneAreaBase();
         network.getGenerator("g1").setMinP(90); // the generator should go down to 70MW to meet the interchange target
         CompletionException thrown = assertThrows(CompletionException.class, () -> loadFlowRunner.run(network, parameters));
         assertEquals("Failed to distribute interchange active power mismatch", thrown.getCause().getMessage());
     }
 
+    static Stream<Arguments> slackDistributionFailureBehaviors() {
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/ac/AreaInterchangeControlTest.java:176]: could be simplified to
return Arrays.stream(OpenLoadFlowParameters.SlackDistributionFailureBehavior.values()).map(Arguments::of);

but seems unused - to be deleted ?

Comment (vmouradian) [reply src/test/java/com/powsybl/openloadflow/ac/AreaInterchangeControlTest.java]: Yes it was unused, removed in [59fd886](https://github.com/powsybl/powsybl-open-loadflow/pull/1099/commits/59fd8860e4c54a93d7506096321ac2b0f04d5d2e)

File: src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java [hunk 2]
```diff
@@ -162,18 +183,32 @@ public DcLoadFlowResult run() {
...
 
         initStateVector(network, equationSystem, new UniformValueVoltageInitializer());
 
-        if (parameters.isDistributedSlack()) {
-            distributeSlack(network, network.getBuses(), parameters.getBalanceType(), parameters.getNetworkParameters().isUseActiveLimits());
+        double slackBusActivePowerMismatch = getActivePowerMismatch(network.getBuses());
+        double distributedActivePower = 0.0;
+        if (parameters.isDistributedSlack() || parameters.isAreaInterchangeControl()) {
+            distributedActivePower = distributeSlack(network, network.getBuses(), parameters.getBalanceType(), parameters.getNetworkParameters().isUseActiveLimits());
         }
```
Comment (jeandemanged) [inline src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java:212]: It was like this before, but here we do not account for slack distribution failure behavior parameter.
Could be in another PR, in this case just add a `// FIXME`.
Important note: in the case of AIC, failing here may not be relevant. Here is done only global slack distribution. But we may have the case of distribution failure when distributing mismatch globally, and at the same time no failure when distributing per area later on in the outerloops.

**Review by jeandemanged — CHANGES_REQUESTED** [review]
Documentation to be updated.

This PR is a quite long, we might want to break it in multiple pieces:
- DC outerloop refactoring
- AreaInterchangeControl

**Comment by vidaldid-rte** [comment]
@jeandemanged  @vmouradian  shouldn't this one be closed now  that it is split in 2 parts ?

**Comment by vmouradian** [comment]
> @jeandemanged @vmouradian shouldn't this one be closed now that it is split in 2 parts ?

Yes, I think we can close it now

**Pull request closed without merging**